jquery.tools/index.d.ts

```diff
@@ -0,0 +1,161 @@
+// Type definitions for jQuery TOOLS 1.2
+// Project: https://github.com/jquerytools/jquerytools
+// Definitions by: Joe Skeen <https://github.com/joeskeen/>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+
+/// <reference types="jquery"/>
+
+declare interface JQuery {
+    overlay(opts?: JQueryTools.overlay.OverlayOptions): JQuery;
+    overlay<T extends JQueryTools.overlay.Overlay>(opts?: JQueryTools.overlay.OverlayOptions): T;
+    data(key: 'overlay'): JQueryTools.overlay.Overlay;
+}
+
+declare interface JQueryStatic {
+    tools: JQueryTools.ToolsStatic;
+}
+
+/** jQuery Tools (http://jquerytools.github.io/documentation/overlay/index.html) */
+declare namespace JQueryTools {
+    interface ToolsStatic {
+        overlay: overlay.OverlayStatic;
+    }
+
+    namespace overlay {
+        interface OverlayStatic {
+            addEffect(effectName: string,
+                      effectFn: (this: Overlay, position: CssOptions, done: () => void) => void,
+                      closeFn: (this: Overlay, done: () => void) => void): void;
+        }
+
+        interface CssOptions {
+            [key: string]: any;
+        }
+
+        interface OverlayOptions {
+            /**
+             * A jQuery selector for the closing elements inside the overlay. These
+             * can be any elements such as links, buttons or images. If this is not
+             * supplied, a close element is auto-generated. Read more about this in
+             * defining close actions.
+             */
+            close?: JQuery;
+            /**
+             * By default, overlays are closed when the mouse is clicked outside the
+             * overlay area. Setting this property to false suppresses this behaviour
+             * which is suitable for modal dialogs.
+             * @default true
+             */
+            closeOnClick?: boolean;
+            /**
+             * By default, overlays are closed when the ESC keyboard key is pressed.
+             * Setting this property to false suppresses this behaviour.
+             * @default true
+             */
+            closeOnEsc?: boolean;
+            /** The speed to close the overlay, in milliseconds */
+            closeSpeed?: number;
+            /**
+             * The effect to be used when an overlay is opened and closed. This can
+             * dramatically change the behaviour of the overlay. By default this tool
+             * uses an effect called "default" which is a simple show/hide effect.
+             * @default 'default'
+             */
+            effect?: 'default' | string;
```

Comment:
@zhengbli I know that as far as type checking goes, `'default' | string` is the same as just `string`, but I was hoping that by providing the string literal in the union type it would provide it as an option for the autocompletion of that option; however, in testing this out I realized that it sadly doesn't work that way.  I'll remove the string literal from both places.